source/base/quadrature_lib.cc

```diff
@@ -1203,8 +1203,9 @@ QSimplex<dim>::QSimplex(const Quadrature<dim> &quad)
   for (unsigned int i = 0; i < quad.size(); ++i)
     {
       double r = 0;
-      for (unsigned int d = 0; d < dim; ++d)
-        r += quad.point(i)[d];
+      if constexpr (dim != 0)
```

Comment:
That's C++17 right? We only require c++14.

Reply:
Good point! I keep forgetting. I'll come up with another solution.